EESSI Governance (#456)
EESSI/docs: docs/governance/policies.md

### 2.5 Optimization Policy

Optimization for performance is strongly encouraged. For compiled software, this means we aim to have the compiler optimize for the supported hardware targets.

---

## 3. Contribution Policies

### 3.1 Software Acceptance

- Contributions of new software are accepted as long as they meet the [Contribution Policy](https://www.eessi.io/docs/adding_software/contribution_policy/).
- The default stance is: **"Yes, unless we can't."**

### 3.2 Non-Software Contributions

- Contributions to build logic, test suites, or infrastructure should be testable and documented.
- Specific requirements may be defined by the relevant Team.

---

## 4. Transparency and Traceability

EESSI aims to be transparent regarding how software was build. A combination of the git history, EasyBuild logs, as well as the `easybuild` subdirectory within every installation (which provides the easyconfig and patch files, easyblock, and any hooks used at installation time) provide a detailed description of how software was configured and built.

---

## 5. Platform Prioritization

**HPC compatibility and performance are prioritized**,

> in cases where technical choices must be made between HPC, Cloud, or PC environments,

Reviewer: ```suggestion
in cases where technical choices must be made between HPC, Cloud, or laptops & workstations,
```

especially for optimization-related decisions.

This does not imply exclusion of other platforms, but reflects the project's primary use case.

---

# EESSI Team Policies

> There are currently no Team Policies yet.

Reviewer 2: ```suggestion
There are no Team Policies yet.
```

- The EESSI software stack version has been archived in _some_ form in which it can still be used by end-users (e.g. as a tarball, an exported CVMFS repository using CernVM-FS Shrinkwrap, or some other form).

A clear runtime error will be printed when an end-user tries to initialize a removed EESSI software stack.

#### 2.4.2 Removal of individual software installations

An individual software installation in EESSI may be removed if:
- It violates licensing terms
- It poses a security risk
- It is no longer maintainable or functional.

A clear runtime error will be printed when an end-user tries to load a module for a removed software installation.

### 2.5 Optimization Policy

Optimization for performance is strongly encouraged. For compiled software, this means we aim to have the compiler optimize for the supported hardware targets.

---

## 3. Contribution Policies

### 3.1 Software Acceptance

- Contributions of new software are accepted as long as they meet the [Contribution Policy](https://www.eessi.io/docs/adding_software/contribution_policy/).
- The default stance is: **"Yes, unless we can't."**

### 3.2 Non-Software Contributions

> - Contributions to build logic, test suites, or infrastructure must be testable and documented.

Reviewer: I would tone this down to "should", since we currently often make changes that are not documented:

```suggestion
- Contributions to build logic, test suites, or infrastructure should be testable and documented.

- Specific requirements may be defined by the relevant Team.

---

## 4. Transparency and Traceability

EESSI aims to be transparent regarding how software was build. A combination of the git history, easybuild build logs, as well as the easybuild subdirectory within every installation (which provides the EasyConfig, EasyBlock, and any hooks used at installation time) provide a detailed description of how software was configured and built.

---

## 5. Platform Prioritization

- In cases where technical choices must be made between HPC, Cloud, or PC environments:
  - **HPC compatibility and performance are prioritized**, especially for optimization-related decisions.
  - This does not imply exclusion of other platforms, but reflects the project's primary use case.

---

# EESSI Team Policies

There are currently no Team Policies yet.

> _Last updated: 23 June 2025_

Reviewer: I would replace this with a "Changlog" section, and mention on top when this page was last updated and point to the changelog for details?

An individual software installation in EESSI may be removed if:
- It violates licensing terms
- It poses a security risk
- It is no longer maintainable or functional.

A clear runtime error will be printed when an end-user tries to load a module for a removed software installation.

### 2.5 Optimization Policy

Optimization for performance is strongly encouraged. For compiled software, this means we aim to have the compiler optimize for the supported hardware targets.

---

## 3. Contribution Policies

### 3.1 Software Acceptance

- Contributions of new software are accepted as long as they meet the [Contribution Policy](https://www.eessi.io/docs/adding_software/contribution_policy/).
- The default stance is: **"Yes, unless we can't."**

### 3.2 Non-Software Contributions

- Contributions to build logic, test suites, or infrastructure must be testable and documented.
- Specific requirements may be defined by the relevant Team.

---

## 4. Transparency and Traceability

### 4.1 Software Bill of Materials (SBOM)

- EESSI is committed to providing a complete SBOM for all deployed software.
- The SBOM should include versioning, licensing, and dependency information.

> - Preferred formats include SPDX or CycloneDX.

Author: I'm not sure if we can already meet the latter two points. If not, does it make sense to include them? I don't think so. Or at least we should make clear it's not currently the case, but is a long term goal.
